src/Voron/Impl/Paging/AbstractPager.cs

```diff
@@ -247,41 +234,80 @@ protected void ThrowOnInvalidPageNumber(long pageNumber)
                 GetSourceName());
         }
 
-        public abstract void ReleaseAllocationInfo(byte* baseAddress, long size);        
+        public abstract void ReleaseAllocationInfo(byte* baseAddress, long size);
 
         // NodeMaxSize - RequiredSpaceForNewNode for 4Kb page is 2038, so we drop this by a bit
         public static readonly int MaxKeySize = 2038 - RequiredSpaceForNewNode;
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool IsKeySizeValid(int keySize)
-        {           
+        {
             if (keySize > MaxKeySize)
                 return false;
 
             return true;
         }
 
-        public void MaybePrefetchMemory(List<long> pagesToPrefetch)
+        protected List<Win32MemoryMapNativeMethods.WIN32_MEMORY_RANGE_ENTRY> SortedPagesToList(List<TreePage> sortedPages)
```

Comment:
@ayende Regarding your remark on this : We want to ask for 10 pages and prefetch 16.  Prefetching will read ahead 7 more pages (31 in linux) every at any case.